writers_rollingfilewriter.go

```diff
@@ -449,7 +461,12 @@ func (rw *rollingFileWriter) Write(bytes []byte) (n int, err error) {
 	// needs to roll if:
 	//   * file roller max file size exceeded OR
 	//   * time roller interval passed
-	nr, err := rw.self.needsToRoll()
+	fi, err := os.Stat(rw.rollingTrackerPath())
+	if err != nil {
+		return 0, err
+	}
+	lastRollTime := fi.ModTime()
```

Comment:
Note the difference between `rw.rollingTrackerPath()` and `rw.originalFileName`: the tracker path is only touched in `rw.createFileAndFolderIfNeeded`, which is only called by `rw.Write` on initialization or on roll.